File: LeetCode-Programs/Minimize Maximum Pair Sum in Array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        m_sum,summ=0,0
        i,j=0,len(nums)-1
        while i<j:
            summ=nums[i]+nums[j]
            m_sum=max(m_sum, summ)
            i=i+1
            j=j-1
        return m_sum

File: LeetCode-Programs/Minimize Maximum Pair Sum in Array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int i=0, j=nums.size()-1,m_sum=0,summ=0;
        while(i<j)
        {
            summ=nums[i]+nums[j];
            m_sum=max(m_sum, summ);
            i++;
            j--;
        }
        return m_sum;
    }
};
